### PR TITLE
feat(unstable): ability to resolve specifiers with no extension, specifiers for a directory, and TS files from JS extensions

### PR DESCRIPTION
**Pull request by dsherret**
This PR adds an `--unstable-sloppy-imports` flag which supports the following for `file:` specifiers:

* Allows writing `./mod` in a specifier to do extension probing.
  - ex. `import { Example } from "./example"` instead of `import { Example } from "./example.ts"`
* Allows writing `./routes` to do directory extension probing for files like `./routes/index.ts`
* Allows writing `./mod.js` for *mod.ts* files.

This functionality is **NOT RECOMMENDED** for general use with Deno:

1. It's not as optimal for perf: https://marvinh.dev/blog/speeding-up-javascript-ecosystem-part-2/
1. It makes tooling in the ecosystem more complex in order to have to understand this.
1. The "Deno way" is to be explicit about what you're doing. It's better in the long run.
1. It doesn't work if published to the Deno registry because doing stuff like extension probing with remote specifiers would be incredibly slow.

This is instead only recommended to help with migrating existing projects to Deno. For example, it's very useful for getting CJS projects written with import/export declaration working in Deno without modifying module specifiers and for supporting TS ESM projects written with `./mod.js` specifiers.

This feature will output warnings to guide the user towards correcting their specifiers. Additionally, Quick fixes are provided in the LSP to update these specifiers:

Shows a quick fix to update the specifier to the proper one

### CODE REVIEW

File: cli/args/flags.rs [hunk 1]
```diff
@@ -447,6 +447,7 @@ pub struct Flags {
   pub unstable: bool,
   pub unstable_bare_node_builtins: bool,
   pub unstable_byonm: bool,
+  pub unstable_loose_imports: bool,
```
Comment (dsherret) [inline cli/args/flags.rs:450]: Alternative name to further discourage use: `--unstable-sloppy-imports`

Comment (bartlomieju) [reply cli/args/flags.rs:450]: I also like `--unstable-sloppy-imports` better.

Comment (dsherret) [reply cli/args/flags.rs]: Updated.

**Comment by dsherret** [comment]
This should be good to go now. I'll add the functionality for the informative error messages when not using this flag in a separate PR.

File: cli/lsp/diagnostics.rs [hunk 2]
```diff
@@ -1193,6 +1196,27 @@ impl DenoDiagnostic {
...
+        } else {
+          format!("./{}", relative)
+        }
+      }
+      None => redirect.to_string(),
+    }
+  } else {
+    redirect.to_string()
+  }
+}
```
Comment (bartlomieju) [inline cli/lsp/diagnostics.rs:1218]: Very cool 👍

File: cli/resolver.rs [hunk 3]
```diff
@@ -371,6 +398,52 @@ impl Resolver for CliGraphResolver {
...
+      format!("(specify path to {} file in directory instead)", file_name)
+    }
+    UnstableSloppyImportsResolution::None(_) => return specifier,
+  };
+  // show a warning when this happens in order to drive
+  // the user towards correcting these specifiers
+  log::warn!(
+    "{} Sloppy import resolution {}\n    at {}",
+    crate::colors::yellow("Warning"),
+    message,
```
Comment (bartlomieju) [inline cli/resolver.rs:436]: Maybe we could do something like: 
`Warning Sloppy import resolution (hint: add .mjs extension)`; where `hint` part is in gray?

**Pull request merged**